.github/workflows/nightly.yml

```diff
@@ -58,3 +58,65 @@ jobs:
+
+    runs-on: self-hosted
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install pip packages
+      run: python3 -m pip install -r third_party/requirements.txt
+
+    - name: Download DPC++
+      run: |
+        wget -O ${{github.workspace}}/dpcpp_compiler.tar.gz https://github.com/intel/llvm/releases/download/sycl-nightly%2F20230626/dpcpp-compiler.tar.gz
+        tar -xvf ${{github.workspace}}/dpcpp_compiler.tar.gz
+
+    - name: Configure CMake
+      run: |
+        export CPATH=/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/
+        export LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws:$LIBRARY_PATH
+        export PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws/llvm-debug/bin/:$PATH
+        export LD_LIBRARY_PATH=/home/u105190/sycl_debug_ws/llvm-debug/lib:$LD_LIBRARY_PATH
+        export C_INCLUDE_PATH=${C_INCLUDE_PATH}:/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/:/lib/x86_64-linux-gnu/lib
+        export CPLUS_INCLUDE_PATH=${CPLUS_INCLUDE_PATH}:/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/:/lib/x86_64-linux-gnu/lib
+        cmake \
+        -B${{github.workspace}}/build \
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}} \
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}} \
+        -DUR_ENABLE_TRACING=ON \
+        -DUR_DEVELOPER_MODE=OFF \
+        -DCMAKE_BUILD_TYPE=${{matrix.build_type}} \
+        -DUR_BUILD_TESTS=ON \
+        -DUR_FORMAT_CPP_STYLE=OFF \
+        -DUR_BUILD_ADAPTER_${{matrix.adapter}}=ON
+
+
+    - name: Build
+      run: |
+        export CPATH=/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/
+        export LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws:$LIBRARY_PATH
+        export PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws/llvm-debug/bin/:$PATH
+        export LD_LIBRARY_PATH=/home/u105190/sycl_debug_ws/llvm-debug/lib:$LD_LIBRARY_PATH
+        export C_INCLUDE_PATH=${C_INCLUDE_PATH}:/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/:/lib/x86_64-linux-gnu/lib
+        export CPLUS_INCLUDE_PATH=${CPLUS_INCLUDE_PATH}:/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/:/lib/x86_64-linux-gnu/lib
+        cmake --build ${{github.workspace}}/build -j $(nproc)
+
+    - name: Test
+      working-directory: ${{github.workspace}}/build
+      run: |
+        CPATH=/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11/ \
+        LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws:$LIBRARY_PATH \
+        PATH=/usr/lib/gcc/x86_64-linux-gnu/11:/home/u105190/sycl_debug_ws/llvm-debug/bin/:$PATH \
+        LD_LIBRARY_PATH=/home/u105190/sycl_debug_ws/llvm-debug/lib:$LD_LIBRARY_PATH \
+        C_INCLUDE_PATH=${C_INCLUDE_PATH}:/lib/x86_64-linux-gnu/lib \
+        CPLUS_INCLUDE_PATH=${C_INCLUDE_PATH}:/lib/x86_64-linux-gnu/lib \
+        ctest -C ${{matrix.build_type}} --output-on-failure -L "conformance" --timeout 3600
```

Comment:
1h timeout is too big (we used this in other workflow for a debugging purposes) - pls make drop this param